route_following_plugin/src/route_following_plugin.cpp

```diff
@@ -133,39 +142,59 @@ namespace route_following_plugin
             if(end_dist < current_progress){
                 break;
             }
+            
+            auto following_lanelets = wm_->getRoute()->followingRelations(shortest_path[last_lanelet_index]);
 
-            resp.new_plan.maneuvers.push_back(
+            //if not already on last lanelet in path, check relation with next lanelet- follow lane change procedure if req, else lane follow
+            if(last_lanelet_index!= (shortest_path.size()-1) && identifyLaneChange(following_lanelets, shortest_path[last_lanelet_index + 1].id()))
+            {
+                //calculate constant start distance for lane change
+                double longl_travel_dist = (target_speed*(LANE_CHANGE_TIME_MAX + buffer_lanechange_time_)); //*config_param for time
+                double lane_change_start_dist;               
+                //lane change start distance needs to be constant                
+                
+                if(wm_->routeTrackPos(shortest_path[last_lanelet_index + 1].centerline2d().back()).downtrack >= route_length){
+                    end_dist = route_length;
+
+                }
+                else{
+                    end_dist  = wm_->routeTrackPos(shortest_path[last_lanelet_index + 1].centerline2d().back()).downtrack;
+                }               
+
+                lane_change_start_dist = current_progress;
+                double starting_lanelet_id = shortest_path[last_lanelet_index].id();
+                double ending_lanelet_id = shortest_path[last_lanelet_index+1].id();
+                resp.new_plan.maneuvers.push_back(
+                composeLaneChangeManeuverMessage(lane_change_start_dist, end_dist, speed_progress, target_speed, 
+                                    starting_lanelet_id, ending_lanelet_id,
+                                    time_progress));
+            }
+            else
+            {
+                resp.new_plan.maneuvers.push_back(
                 composeManeuverMessage(current_progress, end_dist,  
                                     speed_progress, target_speed, 
                                     shortest_path[last_lanelet_index].id(), time_progress));
+                
+            }
             current_progress += dist_diff;
-            time_progress = resp.new_plan.maneuvers.back().lane_following_maneuver.end_time;
+            //time_progress = resp.new_plan.maneuvers.back().lane_following_maneuver.end_time;
```

Comment:
So this time value at the end of each maneuver is now no longer updated? Are we able to plan maneuvers after a lane change?